src/NUnitFramework/framework/Attributes/TestCaseSourceAttribute.cs

```diff
@@ -316,6 +314,27 @@ private static IEnumerable ReturnErrorAsParameter(string errorMessage)
                                                         ", please check the number of parameters passed in the object is correct in the 3rd parameter for the " +
                                                         "TestCaseSourceAttribute and this matches the number of parameters in the target method and try again.";
 
+        private static bool IsCollection(IParameterInfo[] parameters)
+        {
+            return IsColletionOfArrays(parameters) || 
+                   (IsCollectionOfEnumerables(parameters) && !IsCollectionOfStrings(parameters));
```

Comment:
Is the `!IsCollectionOfStrings(parameters)` because we don't want to treat individual strings as collections of characters?